### PR TITLE
Add perf workflow

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
Add initial workflow for performance measurements.

// Preview run for `main` branch: https://github.com/oneapi-src/unified-memory-framework/actions/runs/10996891676/job/30531119551
// manual workflow can be tested only if this workflow is merged.

### CODE REVIEW

**Comment by lukaszstolarczuk** [comment]
thx @kswiecicki - fixed the bug with no tests (I was in the wrong directory).

As for results from benchmarks run on main - as we can see in UR, this scenario is rarely run - we rather want results for specific PR; and of course, there are logs in the CI.

When we have a final version of benchmarks we can tweak this workflow to print some more results; for now we only run basic benchmark just to check if the workflow is running properly.

**Review by bratpiorka — APPROVED** [review]

File: .github/workflows/performance.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        type: number
+        required: true
+
+permissions:
+  contents: read
+  pull-requests: write
+
+env:
+  BUILD_DIR : "${{github.workspace}}/build"
+  INSTL_DIR : "${{github.workspace}}/../install-dir"
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply .github/workflows/performance.yml:18]: Done.

File: .github/workflows/performance.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    # We have to delete the files created outside of using actions.
+    - name: Cleanup self-hosted workspace
+      if: always()
+      run: |
+        ls -la ./
+        rm -rf ./* || true
+
+    - name: Add comment to PR
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      if: ${{ always() && inputs.pr_no != 0 }}
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply .github/workflows/performance.yml:36]: yes, this is needed for GHA to parse it properly

File: .github/workflows/performance.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      with:
+        script: |
+          const pr_no = '${{ inputs.pr_no }}';
+          const provider = 'LEVEL_ZERO';
+          const url = '${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}';
+          const body = `Performance workflow for ${provider}_PROVIDER run:\n${url}`;
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply .github/workflows/performance.yml:42]: this is just a link to the job.
I prefer to omit this extra word, as we have the same sentence in the summary msg

File: .github/workflows/performance.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            markdown = fs.readFileSync('umf_perf_results.md', 'utf8');
+          } catch(err) {
+          }
+
+          const pr_no = '${{ inputs.pr_no }}';
+          const provider = 'LEVEL_ZERO';
+          const url = '${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}';
+          const test_status = '${{ steps.benchmarks.outcome }}';
+          const job_status = '${{ job.status }}';
+          const body = `Performance workflow for ${provider}_PROVIDER run:\n${url}\nJob status: ${job_status}. Test status: ${test_status}.\n ${markdown}`;
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply .github/workflows/performance.yml:110]: the reason we have job status and outcome of `benchmarks` step is exactly to report the results as we wish

also this is just an initial version of the workflow, we'll adjust when we have some proper benchmarks for this workflow.

File: .github/workflows/performance.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        git fetch -- https://github.com/${{github.repository}} +refs/pull/${PR_NO}/*:refs/remotes/origin/pr/${PR_NO}/*
+        git checkout origin/pr/${PR_NO}/merge
+        git rev-parse origin/pr/${PR_NO}/merge
+
+    - name: Configure build
+      run: >
+        cmake
+        -B ${{env.BUILD_DIR}}
+        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_INSTALL_PREFIX="${{env.INSTL_DIR}}"
```
Comment (PatKamin) [inline .github/workflows/performance.yml:72]: I think it's not needed as you don't install UMF.

**Review by PatKamin — CHANGES_REQUESTED** [review]

File: .github/workflows/performance.yml (hunk 5 above)
Comment (lukaszstolarczuk) [reply .github/workflows/performance.yml]: ugh... I tried to be fast, but ended up being just furious... 😄

**Review by PatKamin — APPROVED** [review]

**Pull request merged**